app/controllers/uploads_controller.rb

```diff
@@ -277,7 +276,7 @@ def self.create_upload(
       if url.present? && is_api
         maximum_upload_size = [
           SiteSetting.max_image_size_kb,
-          SiteSetting.max_attachment_size_kb,
+          UploadsController.max_attachment_size_for_user(current_user),
```

Comment:
If we make this a private instance method then we could do

```suggestion
          max_attachment_size_for_user(current_user),
```